pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -44,22 +44,23 @@ type ebpfProgram struct {
 
 func newEBPFProgram(c *config.Config, bpfTelemetry *errtelemetry.EBPFTelemetry) *ebpfProgram {
 	perfHandler := ddebpf.NewPerfHandler(100)
-	mgr := &manager.Manager{
-		PerfMaps: []*manager.PerfMap{
-			{
-				Map: manager.Map{
-					Name: sharedLibrariesPerfMap,
-				},
-				PerfMapOptions: manager.PerfMapOptions{
-					PerfRingBufferSize: 8 * os.Getpagesize(),
-					Watermark:          1,
-					RecordHandler:      perfHandler.RecordHandler,
-					LostHandler:        perfHandler.LostHandler,
-					RecordGetter:       perfHandler.RecordGetter,
-				},
-			},
+	pm := &manager.PerfMap{
+		Map: manager.Map{
+			Name: sharedLibrariesPerfMap,
+		},
+		PerfMapOptions: manager.PerfMapOptions{
+			PerfRingBufferSize: 8 * os.Getpagesize(),
+			Watermark:          1,
+			RecordHandler:      perfHandler.RecordHandler,
+			LostHandler:        perfHandler.LostHandler,
+			RecordGetter:       perfHandler.RecordGetter,
+			TelemetryEnabled:   true,
 		},
 	}
+	mgr := &manager.Manager{
+		PerfMaps: []*manager.PerfMap{pm},
+	}
+	ddebpf.ReportPerfMapTelemetry(pm)
```

Comment:
Suggestion - call it once on the manager instead of adding such a line for every perf-map creation

Reply:
I could add that, but I also wanted it to be opt-in per map.